crates/metal-codegen-llvm/src/ty/function_signature.rs

```diff
@@ -14,7 +14,7 @@ impl CodeGenType for FunctionSignature {
         unsafe {
             let len = self.inputs.len();
 
-            let mut types_to_convert = get_types(llvm, module, &len, &self.inputs);
+            let mut types_to_convert = get_types(llvm, module, &len, self.inputs.values());
```

Comment:
*Technically*, `HashMap` is unordered, which means this can, *technically*, result in different results even with two consecutive, identical calls. `BTreeMap` should be used instead

Reply:
> _Technically_, `HashMap` is unordered, which means this can, _technically_, result in different results even with two consecutive, identical calls. `BTreeMap` should be used instead

Is there an ordered hashmap? I've already caught this before but didn't pin it down to HashMaps thinking it was Ron or serde

Reply:
> [`BTreeMap`](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html) should be used instead